exporter/pulsarexporter/pulsar_exporter.go

```diff
@@ -0,0 +1,202 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pulsarexporter
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/apache/pulsar-client-go/pulsar"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+var errUnrecognizedEncoding = fmt.Errorf("unrecognized encoding")
+
+type PulsarTracesProducer struct {
+	client    pulsar.Client
+	producer  pulsar.Producer
+	topic     string
+	marshaler TracesMarshaler
+	logger    *zap.Logger
+}
+
+func (e *PulsarTracesProducer) tracesPusher(ctx context.Context, td ptrace.Traces) error {
+	messages, err := e.marshaler.Marshal(td, e.topic)
+	if err != nil {
+		return consumererror.NewPermanent(err)
+	}
+
+	for _, message := range messages {
+		_, err = e.producer.Send(ctx, message)
```

Comment:
It's better to use `SendAsync` here to take advantage of batch optimization to get a better write throughput.